### ISSUE TITLE
npm unmet dependencies and missing modules

### ISSUE THREAD

**Issue by winkbrace** [jadjoubran/laravel5-angular-material-starter#2]
When trying to install everything I run into these warnings:

``` bash
[03:58 PM]-[vagrant@packer-virtualbox-iso-1422601639]-[/var/www/angular]
$ sudo chown -R $USER /usr/local

[03:58 PM]-[vagrant@packer-virtualbox-iso-1422601639]-[/var/www/angular]
$ npm install
npm WARN unmet dependency /var/www/angular/node_modules/laravel-elixir requires del@'^0.1.3' but will load
npm WARN unmet dependency /var/www/angular/node_modules/del,
npm WARN unmet dependency which is version 1.1.1
npm WARN unmet dependency /var/www/angular/node_modules/laravel-elixir-imagemin/node_modules/gulp-imagemin/node_modules/imagemin/node_modules/imagemin-gifsicle/node_modules/gifsicle/node_modules/bin-build/node_modules/decompress/node_modules/meow requires object-assign@'^2.0.0' but will load
npm WARN unmet dependency /var/www/angular/node_modules/laravel-elixir-imagemin/node_modules/gulp-imagemin/node_modules/object-assign,
npm WARN unmet dependency which is version 1.0.0
npm WARN unmet dependency /var/www/angular/node_modules/laravel-elixir-imagemin/node_modules/gulp-imagemin/node_modules/imagemin/node_modules/imagemin-gifsicle/node_modules/gifsicle/node_modules/bin-build/node_modules/decompress/node_modules/vinyl-fs requires object-assign@'^2.0.0' but will load
npm WARN unmet dependency /var/www/angular/node_modules/laravel-elixir-imagemin/node_modules/gulp-imagemin/node_modules/object-assign,
npm WARN unmet dependency which is version 1.0.0
npm WARN unmet dependency /var/www/angular/node_modules/laravel-elixir-imagemin/node_modules/gulp-imagemin/node_modules/imagemin/node_modules/imagemin-optipng/node_modules/optipng-bin/node_modules/bin-wrapper/node_modules/bin-check/node_modules/executable requires meow@'^3.1.0' but will load
npm WARN unmet dependency /var/www/angular/node_modules/laravel-elixir-imagemin/node_modules/gulp-imagemin/node_modules/imagemin/node_modules/meow,
npm WARN unmet dependency which is version 2.1.0
npm WARN unmet dependency /var/www/angular/node_modules/laravel-elixir-imagemin/node_modules/imagemin-pngquant/node_modules/pngquant-bin/node_modules/bin-build/node_modules/download/node_modules/request/node_modules/har-validator/node_modules/chalk/node_modules/has-ansi requires get-stdin@'^4.0.1' but will load
npm WARN unmet dependency /var/www/angular/node_modules/laravel-elixir-imagemin/node_modules/imagemin-pngquant/node_modules/pngquant-bin/node_modules/bin-build/node_modules/download/node_modules/get-stdin,
npm WARN unmet dependency which is version 3.0.2
npm WARN unmet dependency /var/www/angular/node_modules/laravel-elixir-imagemin/node_modules/imagemin-pngquant/node_modules/pngquant-bin/node_modules/bin-wrapper/node_modules/download/node_modules/got requires read-all-stream@'^2.0.0' but will load
npm WARN unmet dependency /var/www/angular/node_modules/laravel-elixir-imagemin/node_modules/imagemin-pngquant/node_modules/pngquant-bin/node_modules/bin-wrapper/node_modules/download/node_modules/read-all-stream,
npm WARN unmet dependency which is version 1.0.2
```

I don't know if this is a problem. However when trying to run gulp I run into an error

``` bash
[04:15 PM]-[vagrant@packer-virtualbox-iso-1422601639]-[/var/www/angular]
$ gulp
module.js:338
    throw err;
          ^
Error: Cannot find module 'gulp-util'
    at Function.Module._resolveFilename (module.js:336:15)
    at Function.Module._load (module.js:278:25)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.<anonymous> (/var/www/angular/node_modules/laravel-elixir/Config.js:1:74)
    at Module._compile (module.js:460:26)
    at Object.Module._extensions..js (module.js:478:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)

[04:15 PM]-[vagrant@packer-virtualbox-iso-1422601639]-[/var/www/angular]
$ gulp --version
[16:16:25] CLI version 3.8.11
[16:16:25] Local version 3.8.11

```

I am running this on a freshly installed Ubuntu 14.04 vagrant machine.
I have very little experience with npm and gulp, but googling also doesn't point me in the right direction. I can manually install all missing modules (first gulp-util, then underscore, then probably many more), but I feel there must be another way.

Are these two issues related? Do I need to install or update something?

**Comment by jadjoubran** [comment]
Please try the following:
- manually remove your node_modules folder (at the root directory of the Laravel installation)
- run npm install again
- even if you're getting npm warnings, try to run gulp and if you're still having the same issue let me know

Keep me posted if it works or if you're still having issues

**Comment by winkbrace** [comment]
I solved it. 
First I updated npm itself.
Then I resolved an apparently common issue with npm on vagrant on windows. In windows the max path size is 260 characters, and the recursive path building of npm can create longer paths.
So I had to create a sym-link of node_modules that pointed to a path located outside of the shared folders. 

TLDR; Nothing wrong with your package :smiley:

**Comment by jadjoubran** [comment]
Awesome :D
Let me know if you have any suggestion

**Comment by jadjoubran** [comment]
Oh and check this related package, I think you'll find it useful
https://github.com/jadjoubran/laravel5-generator

**Comment by shiruken1** [comment]
Running a new setup on a LAMP environment, and running into the same issue. Removed node_modules and reran npm install to no avail :(

**Comment by shiruken1** [comment]
Sorry, forgot to include the error I'm getting:

```
$ gulp

Error: Cannot find module 'underscore'
    at Function.Module._resolveFilename (module.js:338:15)
    at Function.Module._load (module.js:280:25)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (/var/www/empowrd/tasks/ngHtml2Js.task.js:12:10)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)

```

**Comment by jadjoubran** [comment]
Are you running inside Vagrant?

**Comment by shiruken1** [comment]
Nope, just regular LAMP (Ubuntu 14.04.3)

**Comment by jadjoubran** [comment]
It turned out that `underscore` is missing from the node dependencies..
you just need to run `npm install underscore --save`
if this command fails then you need to follow these commands in this specific order:

```
rm -rf node_modules
npm install underscore --save
npm install
```

**Comment by shiruken1** [comment]
Cool beans, thanks!
